### PR TITLE
STYLE: Use "typename" for template parameters.

### PR DESCRIPTION
**Pull request by jhlegarreta**
As discussed in:
http://review.source.kitware.com/#/c/12655/

the use of the template keyword "class" was substituted by "typename" in
the toolkit for the reasons stated in that topic.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
@hjmjohnson @fbudin69500 @dzenanz @thewtex Please approve if appropriate and merge. I do not have merge rights.

**Review by dzenanz — APPROVED** [review]

**Pull request merged**
